google-ads/src/main/java/com/google/ads/googleads/lib/callables/ExceptionTransformation.java

```diff
@@ -0,0 +1,7 @@
+package com.google.ads.googleads.lib.callables;
+
+import com.google.api.gax.rpc.ApiException;
+
+public interface ExceptionTransformation {
```

Comment:
Do you anticipate adding more to this interface? If not, you could simply use `java.util.function.Function<ApiException, Throwable>` or, if you'd prefer to have a named interface, just have this `extend java.util.function.Function<ApiException, Throwable>`.

This will also make these objects play nice with other functional programming utilities, etc.

Reply:
I think this interface is important enough to warrant its own definition. We're going to implement this 5 times for each type of RPC the library supports. Passing this around in a Function<> seems like primitive obsession.

Reply:
Could you still have it extend `Function<ApiException, Throwable>`? That'll make it usable with other FP features of Java later, and I don't see a strong reason against it.

Reply:
So this is already compatible with `Function<ApiException, Throwable>`. For example the following code already compiles:

```
    Function<Throwable, Throwable> foo = exceptionTransformation::transform;
```

However, I don't think that this should implement that interface for a couple reasons (1) it's not actually used/needed anywhere, (2) it constrains the future development of this interface to support the functional contract. Happy to change this if there's something I'm missing here?

Reply:
Sounds good. I didn't realize that code snippet you provided will work, which was my main concern.